app/controllers/snipets_controller.rb

```diff
@@ -46,11 +48,9 @@ def create
     @snipet.username = current_user.username
     respond_to do |format|
       if @snipet.save
-        format.html { redirect_to @snipet, notice: 'Snipet was successfully created.' }
-        format.json { render :show, status: :created, location: @snipet }
+        format.html { redirect_to my_snipets_url(current_user.id), notice: 'Snipet was successfully created.' }
```

Comment:
When you have only html as format you do not need to specify it, you your code would look like:

```
if @snipet.save
  redirect_to my_snipets_url(current_user.id), notice: 'Snipet was successfully created.' 
else
  render :new 
end
```